resources/todomvc/big-dom-generator/src/tree-generator.js

```diff
@@ -0,0 +1,73 @@
+import { LCG } from "random-seedable";
+import { DEFAULT_SEED_FOR_RANDOM_NUMBER_GENERATOR, MAX_GENERATED_DOM_DEPTH, MAX_NUMBER_OF_CHILDREN, PROBABILITY_OF_HAVING_CHILDREN, TARGET_SIZE, MIN_NUMBER_OF_MAX_DEPTH_BRANCHES } from "./../params";
+
+const random = new LCG(DEFAULT_SEED_FOR_RANDOM_NUMBER_GENERATOR);
+
+/**
+ * Generates the blueprint of the tree-view side panel for the complex DOM shell.
+ * It starts with the minimum number of maximum-depth branches and randomly adds
+ * children to the nodes in a breadth-first manner.
+ * The weight parameters represent how many DOM nodes are generated for each type of node.
+ **/
+export const generateTreeHead = ({ listWeight, openItemWeight, closedItemWeight }) => {
```

Comment:
- OpenItem is meant to be an expandable item and closedItem a non-expandable item, we are renaming these. 
- You are right, when rendering we use the number of children to decide if this item will be a todo-list or a sprint-folder, it would be clearer to use the type since it is already there.
- Not every closedItem is replaced by an openItem, all items start as closed and are mark open if the algorithm adds children, we can say this in a comment.